gk/main.c

```diff
@@ -294,6 +310,7 @@ gk_process_request(struct flow_entry *fe, struct ipacket *packet,
 	} else {
 		fe->u.request.last_priority = priority;
 		fe->u.request.allowance = START_ALLOWANCE - 1;
+		fe->expire_at = calc_request_expire_at(now);
```

Comment:
We can't use `calc_initial_request_expiration()` here. The calculation should be done as follows:
```
uint8_t above_priority = priority + 3;
if (unlikely(above_priority >= sizeof(uint64_t) * 8)) {
	/* Avoid overflow of the left shift operator (i.e. <<) in the `else` body. */
	fe->expire_at = UINT64_MAX;
} else {
	/*
	 * TCP waits for 2^i seconds to retransmit a SYN packet; The variable i is greater or equal to 0.
	 * Thus, the corresponding priority p is floor(log_2(2^i * 10^12)) = foor(i + log_2(10^12)) = i + 39.
	 * If one sets above_priority = p + 3 and waits for the amount of time corresponding for the above_priority priority,
	 * TCP can transmit two more SYN packets:
	 * 2^(i+2) * 10^12 <= 2 ^ above_priority =>
	 * 2^(i+2) * 10^12 <= 2^(i+2) * 2^40 =>
	 * 10^12 <= 2^40 (TRUE)
	 */

	/* The cast `(uint64_t)` is needed to force the compiler to use the 64-bit version of `<<`. */
	fe->expire_at = now + (uint64_t)1 << above_priority;
}
```
Feel free to reformat the comments.

Reply:
Fixing the proof:
```
	 * TCP can transmit two more SYN packets:
	 * (2^(i+1)+2^(i+2)) * 10^12 <= 2 ^ above_priority =>
	 * 2^(i+1) * 10^12 + 2^(i+2) * 10^12 <= 2^(i+2) * 2^41 =>
	 * 2^(i+2) * 5 * 10^11 + 2^(i+2) * 10^12 <= 2^(i+2) * 2^41 =>
	 * 5 * 10^11 + 10^12 <= 2^41 (TRUE)
	 */
```
In order to fix the proof, I had to make `above_priority = priority + 4`, so align the code and comment with this change.